nodeup/pkg/model/cloudconfig.go

```diff
@@ -38,6 +44,31 @@ func (b *CloudConfigBuilder) Build(c *fi.ModelBuilderContext) error {
 	var lines []string
 
 	cloudConfig := b.Cluster.Spec.CloudConfig
+
+	if fi.CloudProviderID(b.Cluster.Spec.CloudProvider) == fi.CloudProviderVSphere {
+		actualKubernetesVersion, err := util.ParseKubernetesVersion(b.Cluster.Spec.KubernetesVersion)
```

Comment:
This whole functionality of reading VM UUID deserves a separate method. Otherwise its unnecessarily increasing length of Build function.